src/test/compile-fail/associated-const-type-parameter-arrays.rs

```diff
@@ -26,7 +26,9 @@ impl Foo for Def {
 
 pub fn test<A: Foo, B: Foo>() {
     let _array: [u32; <A as Foo>::Y];
-    //~^ ERROR the trait bound `A: Foo` is not satisfied
+    //~^ ERROR cannot use an outer type parameter in this context [E0402]
```

Comment:
Is there a backwards compatibility concern here?  Probably not, since this sort of doesn't work now -- but I feel like it *should* work someday, so it seems odd to be considering `A` as an "outer" type parameter.